src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,158 @@
+package de.tum.in.www1.artemis.service;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import org.apache.commons.fileupload.*;
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.apache.pdfbox.multipdf.Splitter;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.service.util.Tuple;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitSplitDTO;
+
+@Service
+public class LectureUnitProcessingService {
+
+    /**
+     * Split units from given file according to given split information.
+     * @param lectureUnitSplitDTOs The split information
+     * @param file The file (lecture slide) to be split
+     * @return The prepared units to be saved
+     */
+    public List<LectureUnitDTO> splitUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, MultipartFile file) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        List<LectureUnitDTO> units = new ArrayList<>();
+        Splitter pdfSplitter = new Splitter();
+        PDDocument document = PDDocument.load(file.getBytes());
+
+        for (LectureUnitSplitDTO lectureUnit : lectureUnitSplitDTOs) {
+            AttachmentUnit attachmentUnit = new AttachmentUnit();
+            Attachment attachment = new Attachment();
+            PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+            pdfSplitter.setStartPage(lectureUnit.startPage());
+            pdfSplitter.setEndPage(lectureUnit.endPage());
+            pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+            List<PDDocument> documentUnits = pdfSplitter.split(document);
+            pdDocumentInformation.setTitle(lectureUnit.unitName());
+            documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
```

Comment:
You are splitting the document again and again, but at least going by the return value, you should be able to define all the places to split at above the loop and then work with the result. Calling split() every time seems like unnecessary overhead to me.

Reply:
I am not sure if that is possible to do, because units can have different page length and split() function will split it equally (e.g. if file is 100 pages and i set start page to 1 and end page to 20 it will split in 5 documents). Its not possible to set different split settings in one go.